src/modm/driver/inertial/bmi088_transport_impl.hpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2023, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_BMI088_TRANSPORT_HPP
+#error "Don't include this file directly, use 'bmi088_transport.hpp' instead!"
+#endif
+
+#include <modm/processing/fiber.hpp>
+
+namespace modm
+{
+
+template<typename SpiMaster, typename AccCs, typename GyroCs>
+void
+Bmi088SpiTransport<SpiMaster, AccCs, GyroCs>::initialize()
+{
+	// rising edge on CS required to enable accelerometer SPI mode
+	AccCs::setOutput(false);
+	modm::delay_ns(100);
+	AccCs::setOutput(true);
+
+	GyroCs::setOutput(true);
+}
+
+template<typename SpiMaster, typename AccCs, typename GyroCs>
+std::span<uint8_t>
+Bmi088SpiTransport<SpiMaster, AccCs, GyroCs>::readRegisters(AccRegister startReg,
+										uint8_t count)
+{
+	return readRegisters<AccCs>(static_cast<uint8_t>(startReg), count, true);
+}
+
+template<typename SpiMaster, typename AccCs, typename GyroCs>
+std::span<uint8_t>
+Bmi088SpiTransport<SpiMaster, AccCs, GyroCs>::readRegisters(GyroRegister startReg,
+										uint8_t count)
+{
+	return readRegisters<GyroCs>(static_cast<uint8_t>(startReg), count, false);
+}
+
+template<typename SpiMaster, typename AccCs, typename GyroCs>
+template<typename Cs>
+std::span<uint8_t>
+Bmi088SpiTransport<SpiMaster, AccCs, GyroCs>::readRegisters(uint8_t startReg,
+										uint8_t count, bool dummyByte)
+{
+	if (count > MaxRegisterSequence) {
+		return {};
+	}
+
+	while (!this->acquireMaster()) {
+		modm::fiber::yield();
+	}
+	Cs::reset();
```

Comment:
Something like 
```cpp
auto lock = SpiLock<Cs>();
```
would aquire SPI + reset CS and release SPI + set CS on destruction?

Reply:
That's `SpiDevice::acquireMaster()`, not `SpiMaster::acquire()` which I have cleaned up. 
The refactoring is certainly worth considering but I'd consider it out of scope for this PR.